src/build-disk-image.ts

```diff
@@ -104,61 +103,68 @@ export async function buildDiskImage(image: any) {
+        // Step 2. Check if there are any previous builds and remove them
         progress.report({ increment: 5 });
+        await containerUtils.removeContainerIfExists(buildImageContainer.name);
 
-        const containerId = await createImage(image, selectedType, selectedFolder, imagePath);
+        // Step 3. Create and start the container for the actual build
         progress.report({ increment: 6 });
+        const containerId = await containerUtils.createAndStartContainer(image.engineId, buildImageContainer);
 
+        // Since we have started the container, we can now go get the logs
         await logContainer(image, containerId, progress, data => {
           logData += data;
           console.log('log:' + logData);
         });
 
-        // Wait for container to exit so that the task doesn't end and we can monitor progress
-        let containerRunning = true;
-        while (containerRunning) {
-          await extensionApi.containerEngine.listContainers().then(containers => {
-            containers.forEach(container => {
-              // check if container is stopped
-              if (container.Id === containerId && container.State === 'exited') {
-                // remove the container
-                // Cant do this until we extract the logs
-                //extensionApi.containerEngine.deleteContainer(image.engineId, containerId);
-                containerRunning = false;
-              }
-            });
-          });
-          await new Promise(r => setTimeout(r, 1000));
-        }
+        // Wait for the container to exit
+        // This function will ensure it exits with a zero exit code
+        // if it does not, it will error out.
+        progress.report({ increment: 7 });
+        await containerUtils.waitForContainerToExit(buildImageContainer.name);
 
+        // If we get here, the container has exited with a zero exit code
+        // it's successful as well so we will write the log file
         successful = true;
-
-        fs.writeFileSync(logPath, logData, { flag: 'w' });
         telemetryData.success = true;
       } catch (error) {
         console.error(error);
+        telemetryData.error = error;
+        await extensionApi.window.showErrorMessage(`Unable to build disk image: ${error}. Check logs at ${logPath}`);
+      } finally {
+        // Regardless, write the log file and ignore if we can't even write it.
         try {
           fs.writeFileSync(logPath, logData, { flag: 'w' });
         } catch (e) {
           // ignore
         }
-        telemetryData.error = error;
-        await extensionApi.window.showErrorMessage(`Unable to build disk image: ${error}. Check logs at ${logPath}`);
       }
 
       // Mark the task as completed
       progress.report({ increment: -1 });
-
       telemetryLogger.logUsage('buildDiskImage', telemetryData);
 
-      // Only if success = true and type = ami
-      if ((successful && selectedType === 'ami') || selectedType === 'raw') {
```

Comment:
See if you can spot the logic error / hack for the demo haha. This part I fixed up since it kept saying success if you did raw...